relation setting failure

```
Jun  7 19:55:14 tupac conjure-up/bundle-observable-kubernetes: [ERROR] Showing dialog for exception:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/conjure_up-2.0.0-py3.5.egg/conjure/juju.py", line 300, in do_add_all
    params=params)
  File "/usr/local/lib/python3.5/dist-packages/conjure_up-2.0.0-py3.5.egg/macumba/v2.py", line 100, in _request
    'Params': params})
  File "/usr/local/lib/python3.5/dist-packages/conjure_up-2.0.0-py3.5.egg/macumba/v2.py", line 111, in call
    return self.receive(req_id, timeout)
  File "/usr/local/lib/python3.5/dist-packages/conjure_up-2.0.0-py3.5.egg/macumba/api.py", line 126, in receive
    raise ServerError(res['Error'], res)
macumba.errors.ServerError: (ServerError(...), 'cannot add relation "filebeat:elasticsearch elasticsearch:client": relation already exists')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.5/concurrent/futures/thread.py", line 55, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.5/dist-packages/conjure_up-2.0.0-py3.5.egg/conjure/juju.py", line 30, in _decorator
    return f(*args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/conjure_up-2.0.0-py3.5.egg/conjure/juju.py", line 303, in do_add_all
    exc_cb(e)
  File "/usr/local/lib/python3.5/dist-packages/conjure_up-2.0.0-py3.5.egg/conjure/controllers/deploy/gui.py", line 26, in __handle_exception
    app.ui.show_exception_message(exc)
  File "/usr/local/lib/python3.5/dist-packages/conjure_up-2.0.0-py3.5.egg/conjure/ui/__init__.py", line 17, in show_exception_message
    self.frame.body = ErrorView(ex.args[0])
  File "/usr/local/lib/python3.5/dist-packages/conjure_up-2.0.0-py3.5.egg/ubuntui/views/error.py", line 23, in __init__
    Padding.center_80(Color.error_major(Text(error))),
  File "/usr/lib/python3/dist-packages/urwid/widget.py", line 836, in __init__
    self.set_text(markup)
  File "/usr/lib/python3/dist-packages/urwid/widget.py", line 879, in set_text
    self._text, self._attrib = decompose_tagmarkup(markup)
  File "/usr/lib/python3/dist-packages/urwid/util.py", line 375, in decompose_tagmarkup
    tl, al = _tagmarkup_recurse(tm, None)
  File "/usr/lib/python3/dist-packages/urwid/util.py", line 416, in _tagmarkup_recurse
    raise TagMarkupException("Invalid markup element: %r" % tm)
urwid.util.TagMarkupException: Invalid markup element: ServerError(ServerError(...), 'cannot add relation "filebeat:elasticsearch elasticsearch:client": relation already exists')
```

With latest juju ops queue the relations seem to be duplicating and this error is not handled properly it looks like

--- a/conjure/juju.py
+++ b/conjure/juju.py
@@ -289,19 +289,23 @@ def set_relations(services, exc_cb=None):
     Arguments:
     services: list of services with relations to set
     """
+    relations = set()
+    for service in services:
+        for a, b in service.relations:
+            if (a, b) not in relations and (b, a) not in relations:
+                relations.add((a, b))
 
     @requires_login
     def do_add_all():
-        for service in services:
-            for a, b in service.relations:
-                params = {"Endpoints": [a, b]}
-                try:
-                    this.CLIENT.Service(request="AddRelation",
-                                        params=params)
-                except Exception as e:
-                    if exc_cb:
-                        exc_cb(e)
-                    return
+        for a, b in list(relations):
+            params = {"Endpoints": [a, b]}
+            try:
+                this.CLIENT.Service(request="AddRelation",
+                                    params=params)
+            except Exception as e:
+                if exc_cb:
+                    exc_cb(e)
+                return
 
     return async.submit(do_add_all,
                         exc_cb,

